### ISSUE TITLE
Readiness & Liveness probe failed

### ISSUE THREAD

**Issue by FernandoMiguel** [kubecost/cost-analyzer-helm-chart#2422]
Thanks for your interest in Kubecost! To ensure you get the help you need, please follow these guidelines:

- If you have questions about product functionality/usage or need assistance with product installation/configuration, please email <support@kubecost.com>, where our support engineering team will be happy to help!
- If you have a reproducible bug, you're in the right place. Please include the information requested below.

---

**Describe the bug**  
after a minute or so of deploying 1.104.* , the cost-analyzer goes unhealthy, showing issues with Readiness & Liveness probe failing

**To Reproduce**  

1. we deploy with argocd
2. access the ingress endpoint
3. See error

**Expected behavior**  
for the pod to pass checks and stay online

**Screenshots**  
![image](https://github.com/kubecost/cost-analyzer-helm-chart/assets/168366/99bd0bab-6c05-4571-8ebc-b1cc2da2c055)
**What impact will this have on your ability to get value out of Kubecost?**

tail of log
```
│ 2023-07-07T14:27:05.221236366Z ERR unable to get most recent valid asset set: could not obtain latest valid asset set
│ 2023-07-07T14:27:05.221250276Z WRN got error could not obtain latest valid asset set for metric localDisks%%Production, not adding to cache
│ 2023-07-07T14:27:05.244470351Z INF Found Discount for InstanceType: c5.xlarge of 0.00
│ 2023-07-07T14:27:05.244504229Z INF Found Discount for InstanceType: c6i.large of 0.00
│ 2023-07-07T14:27:05.244521959Z INF Found Discount for InstanceType: c6i.large of 0.00
│ 2023-07-07T14:27:05.244544373Z INF Found Discount for InstanceType: c5.xlarge of 0.00
│ 2023-07-07T14:27:05.325602928Z INF FA[*types.ContainerStatsSet]: Successfully built and inserted window '{Start:2023-04-10 00:00:00 +0000 UTC End:2023-04-11 00:00:00 +0000 UTC}' in 180.661141ms
│ 2023-07-07T14:27:05.325634915Z INF FA[*types.ContainerStatsSet]: Building window: {Start:2023-04-09 00:00:00 +0000 UTC End:2023-04-10 00:00:00 +0000 UTC}
│ 2023-07-07T14:27:05.473824025Z INF FA[*types.ContainerStatsSet]: Successfully built and inserted window '{Start:2023-04-09 00:00:00 +0000 UTC End:2023-04-10 00:00:00 +0000 UTC}' in 148.176187ms
│ 2023-07-07T14:27:05.473917744Z INF FA[*types.ContainerStatsSet]: Building window: {Start:2023-04-08 00:00:00 +0000 UTC End:2023-04-09 00:00:00 +0000 UTC}
│ 2023-07-07T14:27:05.61419719Z INF FA[*types.ContainerStatsSet]: Successfully built and inserted window '{Start:2023-04-08 00:00:00 +0000 UTC End:2023-04-09 00:00:00 +0000 UTC}' in 140.27244ms
│ 2023-07-07T14:27:05.614234199Z INF FA[*types.ContainerStatsSet]: Building window: {Start:2023-04-07 00:00:00 +0000 UTC End:2023-04-08 00:00:00 +0000 UTC}
│ 2023-07-07T14:27:05.761350193Z INF FA[*types.ContainerStatsSet]: Successfully built and inserted window '{Start:2023-04-07 00:00:00 +0000 UTC End:2023-04-08 00:00:00 +0000 UTC}' in 147.105371ms
│ 2023-07-07T14:27:19.528233427Z INF ETL: Allocation: QueryAllocation([2023-07-06T14:27:19+0000, 2023-07-07T14:27:19+0000), [cluster node namespace pod container]) from StoreDriver[1d] 1.312527ms [query 839.448µs] [idle/tenancy 463ns] [external 179ns] [aggregate 471.45µs] [accumulate 162ns] [stop 825ns]│
│ 2023-07-07T14:27:19.529835671Z INF ETL: Allocation: QueryAllocation([2023-07-06T14:27:19+0000, 2023-07-07T14:27:19+0000), [cluster node namespace pod container]) from StoreDriver[1d] 1.233042ms [query 714.963µs] [idle/tenancy 391ns] [external 166ns] [aggregate 517.189µs] [accumulate 230ns] [stop 103ns]
│ 2023-07-07T14:27:19.530794527Z INF ETL: Asset: QueryAsset([2023-07-06T14:27:19+0000, 2023-07-07T14:27:19+0000), []) from StoreDriver[1h] 774.04µs [query 605.548µs] [cloud 16.617µs] [aggregate 151.629µs] [accumulate 137ns] [stop 109ns]│
│ 2023-07-07T14:27:19.533298924Z INF ETL: Asset: QueryAsset([2023-07-07T08:27:19+0000, 2023-07-07T14:27:19+0000), []) from StoreDriver[1h] 272.835µs [query 233.921µs] [cloud 4.65µs] [aggregate 33.938µs] [accumulate 216ns] [stop 110ns]│
│ 2023-07-07T14:27:19.534573079Z INF ETL: Allocation: QueryAllocation([2023-07-05T14:27:19+0000, 2023-07-07T14:27:19+0000), [cluster node namespace pod container]) from StoreDriver[1d] 1.210133ms [query 894.611µs] [idle/tenancy 318ns] [external 101ns] [aggregate 314.833µs] [accumulate 155ns] [stop 115ns]
│ 2023-07-07T14:27:19.535317024Z INF ETL: Asset: QueryAsset([2023-07-05T14:27:19+0000, 2023-07-07T14:27:19+0000), [cluster]) from StoreDriver[1d] 206.991µs [query 148.971µs] [cloud 2.528µs] [aggregate 43.082µs] [accumulate 12.26µs] [stop 150ns]│
│ 2023-07-07T14:27:19.536048566Z INF ETL: Asset: QueryAsset([2023-07-06T14:27:19+0000, 2023-07-07T14:27:19+0000), []) from StoreDriver[1h] 681.977µs [query 546.391µs] [cloud 17.426µs] [aggregate 117.488µs] [accumulate 185ns] [stop 487ns]
│ 2023-07-07T14:27:19.53875663Z INF ETL: Allocation: QueryAllocation([2023-07-05T14:27:19+0000, 2023-07-07T14:27:19+0000), [cluster node namespace pod container]) from StoreDriver[1d] 1.155957ms [query 865.88µs] [idle/tenancy 410ns] [external 172ns] [aggregate 289.185µs] [accumulate 177ns] [stop 133ns]
│ 2023-07-07T14:27:19.543074459Z INF ETL: Allocation: QueryAllocation([2023-07-05T14:27:19+0000, 2023-07-07T14:27:19+0000), [cluster node namespace pod controller]) from StoreDriver[1d] 1.908696ms [query 817.416µs] [idle/tenancy 480ns] [external 174ns] [aggregate 633.667µs] [accumulate 456.725µs] [stop 234ns]│
│ 2023-07-07T14:27:19.543517862Z INF ETL: Asset: QueryAsset([2023-07-07T08:27:19+0000, 2023-07-07T14:27:19+0000), []) from StoreDriver[1h] 254.065µs [query 216.252µs] [cloud 7.315µs] [aggregate 30.215µs] [accumulate 180ns] [stop 103ns]
│ 2023-07-07T14:27:19.543554036Z ERR unable to get most recent valid asset set: could not obtain latest valid asset set
│ 2023-07-07T14:27:19.54356922Z WRN got error could not obtain latest valid asset set for metric localDisks%%High-Availability, not adding to cache
│ 2023-07-07T14:27:19.564373506Z INF Found Discount for InstanceType: c6i.large of 0.00
│ 2023-07-07T14:27:19.564410848Z INF Found Discount for InstanceType: c6i.large of 0.00
│ 2023-07-07T14:27:19.564430135Z INF Found Discount for InstanceType: c5.xlarge of 0.00
│ 2023-07-07T14:27:19.564448586Z INF Found Discount for InstanceType: c5.xlarge of 0.00
│ Stream closed EOF for kubecost/cluster-kubecost-cost-analyzer-c5f9cf767-gkrhg (cost-model)
``` 

**Bug report**
Please generate a bug report and attach the resulting `kubecost-bug-report.txt` here:
[kubecost-bug-report (1).txt](https://github.com/kubecost/cost-analyzer-helm-chart/files/11983017/kubecost-bug-report.1.txt)

**Comment by kirbsauce** [comment]
@FernandoMiguel , thanks for the report!  We'll review and come back with next steps.

**Comment by FernandoMiguel** [comment]
This has been broken for all Releases of 1.104 and 1.105

Exact same config with 1.103 works fine, but sadly Athena access cross account is broken. That has been fixed in a early release of 1.104, hence why we been trying to upgrade to it, but been getting stuck on the probe issue or some other underlying issue.
Let me know if we can provide any more details or a redacted version of the values.yaml (although one should be in Zendesk).

**Comment by kirbsauce** [comment]
Thanks @FernandoMiguel .  I'm not having any luck finding a related zendesk ticket.  Do you have a ticket number?

**Comment by FernandoMiguel** [comment]
Request (4395) among others from the same email

**Comment by kirbsauce** [comment]
My apologies, @FernandoMiguel , I now realize you have already worked with support on this.  @kwombach12 , support believes this is something in need of engineering attention.

**Comment by kwombach12** [comment]
@kirbsauce Thanks for investigating! @FernandoMiguel I will have engineering take a look!

**Comment by 55TFSI** [comment]
I am facing the same issue.. using argocd. 
Any version I can use for a POC?

**Comment by FernandoMiguel** [comment]
@55TFSI i was told by support version 105 should address these issues.
I'll try to update this card once I get the opportunity to test

**Comment by FernandoMiguel** [comment]
just tested prod-1.105.0-rc.7 and still fails in the same way

**Comment by rossfisherkc** [comment]
Sorry to hear about the failure!
​
We have this bugfix scheduled for 1.106. I'm unable to find mention of 1.105 version specifically being targeted in the case (https://kubecost.zendesk.com/agent/tickets/4395), however, apologies for any confusion or miscommunication there.

For the time being, I'd be interested to know if Kubecost works successfully without Argo-CD to rule it out, or if it's the specific failure mode cc @55TFSI

**Comment by 55TFSI** [comment]
Hello, thank you so much, 
I successfully set it up using helm manually without argocd.

CC: @rossfisherkc @FernandoMiguel

**Comment by FernandoMiguel** [comment]
I also tried via helm cli.
that led us to discover something that we were not able to see when deploying with argo.
I'm ashamed to say, when athena files get processed, it balloons the memory usage and gets OOMKilled. we were using the chart default values (256MBs) as limit, and it needs over 700MBs.
We have since increased it to 1GB and has been stable running latest version 1.105 rc10.
Thanks all for the help and support, this was a tricky one to spot since it would almost immediately go into crashloop and we were missing the OOMKill.
Maybe conside having a note in the docs that if athena is in use, more memory resources will be required.
This ticket can now be closed.
